src/vimscript/pattern.ts

```diff
@@ -115,6 +116,11 @@ export class Pattern {
     while (true) {
       const match = this.regex.exec(haystack);
 
+      let searchTimeout = false;
+      setTimeout(() => {
+        searchTimeout = true;
```

Comment:
Aha! I tried doing this once but was thinking about it wrong (I don't have much experience with js/node) and got into a classic XY problem googling "how to cancel a promise nodejs", which turns out to be hard or impossible (I forget). Turns out to be so simple 🙃